src/flows/components/panel/Results.tsx

```diff
@@ -19,6 +18,165 @@ import {event} from 'src/cloud/utils/reporting'
 import {RemoteDataState} from 'src/types'
 import {Visibility} from 'src/types/flows'
 
+const HEADER_HEIGHT = 51
+const ROW_HEIGHT = 25
+
+interface TableProps {
+  startRow: number
+  results: FluxResult
+  height: number
+  pageSize: number
+  setPageSize: (size: number) => void
+}
+
+const ResultTable: FC<TableProps> = ({
+  startRow,
+  results,
+  height,
+  pageSize,
+  setPageSize,
+}) => {
+  const kids = useMemo(() => {
+    if (!height) {
+      return
+    }
+
+    let runningHeight = 60 // this is to account for the space around the table
+    let rowIdx = startRow
+    let currentTable
+
+    while (startRow <= results.parsed.table.length) {
+      if (results.parsed.table.columns.table.data[rowIdx] !== currentTable) {
```

Comment:
whew! This is a very confident string of object introspection lol. How confident are you that results.parsed.table.columns.table.data will always exist?

Reply:
these are always famous last words, but this component only renders results.parsed, and if it's gotten that far, i'm pretty sure that table.colums.table will exist until we swap out the parser